facebook/OpenBIC
# ol2: op: Fix E1.S PERST gpio number

# Description:
1. Revise E1.S PERST gpio number to meet the mechanical design
2. Distinguish EXP A and EXP B E1.S present pin GPIO number.
3. Distinguish EXP A and EXP B E1.S power enable and power good ping GPIO number.

# Motivation:
1. To fix power off a E1.S device will cause other E1.S device link down.
2. To fix hotplug E1.S won't repower on.
3. To fix BIC will send E1.S power fault SEL by mistake.

# Test Plan:
1. Build and test pass on OLP2.0 system. pass

2. Check the host message while doing power control on E1.S device. Before fix:
root@bmc-oob:~# power-util slot1 2U-dev0 off
Powering fru 1 dev 2U-dev0 to OFF state...

[root@PureCent9OS ~]# [   53.040455] pcieport 0000:c0:01.6: pciehp: Slot(5): Link Down
[   53.067101] nvme5n1: detected capacity change from 15002931888 to 0
[   53.100614] pcieport 0000:c0:01.6: pciehp: Slot(5): Card present
[   53.171604] pcieport 0000:c0:01.4: pciehp: Slot(3): Link Down
[   53.198106] nvme3n1: detected capacity change from 15002931888 to 0
[   53.235291] pcieport 0000:c0:01.4: pciehp: Slot(3): Card present
[   54.335531] pcieport 0000:c0:01.6: pciehp: Slot(5): No link
[   54.470540] pcieport 0000:c0:01.4: pciehp: Slot(3): No link

After fix:
root@bmc-oob:~# power-util slot1 2U-dev0 off
Powering fru 1 dev 2U-dev0 to OFF state...

[root@PureCent9OS ~]# [  358.194544] pcieport 0000:c0:01.4: pciehp: Slot(3): Link Down
[  358.221648] nvme3n1: detected capacity change from 15002931888 to 0
[  358.257780] pcieport 0000:c0:01.4: pciehp: Slot(3): Card present
[  359.493090] pcieport 0000:c0:01.4: pciehp: Slot(3): No link

3. Check the E1.S device power status by its LED after hot plug. pass

4. Check SEL after doing sled-cycle. root@bmc-oob:~# log-util all --print
2018 Mar 09 05:54:54 log-util: User cleared all logs
0    all      2018-03-09 05:55:00    power-util       SLED_CYCLE starting...
1    slot1    2018-03-09 04:35:51    power-util       SERVER_POWER_ON successful for FRU: 1
0    all      2018-03-09 04:35:52    gpiointrd        ASSERT: GPIOH4 - PRSNT_MB_BMC_SLOT1_BB_N
0    all      2018-03-09 04:35:52    gpiointrd        slot1 present
1    slot1    2018-03-09 04:35:52    gpiointrd        FRU: 1, CPU presence
6    nic      2018-03-09 04:35:52    ncsid            FRU: 6 NIC AEN Supported: 0x70207, AEN Enable Mask=0x70007
6    nic      2018-03-09 04:35:52    ncsid            FRU: 6 PLDM type supported = 0x25
6    nic      2018-03-09 04:35:52    ncsid            FRU: 6 PLDM type 0 version = 1.0.0.0
6    nic      2018-03-09 04:35:52    ncsid            FRU: 6 PLDM type 2 version = 1.1.1.0
6    nic      2018-03-09 04:35:52    ncsid            FRU: 6 PLDM type 5 version = 1.0.0.0
0    all      2018-03-09 04:35:52    healthd          ASSERT: Verified boot failure (3,35)
0    all      2018-03-09 04:35:52    healthd          Verified boot failure reason: U-Boot FIT did not contain the /keys node
0    all      2018-03-09 04:35:52    healthd          SLED Powered OFF at Fri Mar  9 05:55:00 2018
7    bmc      2018-03-09 04:36:19    sensord          ASSERT: Upper Non Critical threshold - raised - FRU: 7, num: 0xE0 curr_val: 13786.00 RPM, thresh_val: 13500.00 RPM, snr: BB_FAN0_TACH_RPM
7    bmc      2018-03-09 04:36:19    sensord          ASSERT: Upper Non Critical threshold - raised - FRU: 7, num: 0xE1 curr_val: 13563.00 RPM, thresh_val: 13500.00 RPM, snr: BB_FAN1_TACH_RPM
7    bmc      2018-03-09 04:36:19    sensord          ASSERT: Upper Non Critical threshold - raised - FRU: 7, num: 0xE2 curr_val: 13690.00 RPM, thresh_val: 13500.00 RPM, snr: BB_FAN2_TACH_RPM
7    bmc      2018-03-09 04:36:19    sensord          ASSERT: Upper Non Critical threshold - raised - FRU: 7, num: 0xE3 curr_val: 13786.00 RPM, thresh_val: 13500.00 RPM, snr: BB_FAN3_TACH_RPM
1    slot1    2018-03-09 04:37:02    ipmid            SEL Entry: FRU: 1, Record: Facebook Unified SEL (0xFB), GeneralInfo: POST(0x28), POST Failure Event: BIOS fails to get the certificate from BMC, Failure Detail: No certificate at BMC
0    all      2018-03-09 04:37:31    healthd          SLED Powered ON at Fri Mar  9 04:34:49 2018
1    slot1    2018-03-09 04:37:34    ipmid            SEL Entry: FRU: 1, Record: Facebook Unified SEL (0xFB), GeneralInfo: POST(0x28), POST Failure Event: System HTTP boot fail, Fail Type: IPv6 fail, Error Code: 0x15
1    slot1    2018-03-09 04:37:39    ipmid            SEL Entry: FRU: 1, Record: Facebook Unified SEL (0xFB), GeneralInfo: POST(0x28), POST Failure Event: System HTTP boot fail, Fail Type: IPv6 fail, Error Code: 0x15
1    slot1    2018-03-09 04:37:45    ipmid            SEL Entry: FRU: 1, Record: Facebook Unified SEL (0xFB), GeneralInfo: POST(0x28), POST Failure Event: System PXE boot fail, Fail Type: IPv6 fail, Error Code: 0x15
